tools/cooja/apps/avrora/build.gradle

```diff
@@ -0,0 +1,24 @@
+apply plugin: "java"
+
+jar.baseName = 'cooja_avrora'
+
+repositories {
+  jcenter()
+}
+
+sourceSets {
+  main {
+    java {
+      srcDir 'src'
+    }
+  }
+
+}
+
+
+dependencies {
+    compile project(":")
+    compile files("lib/avrora-small.jar")
```

Comment:
You are mixing up a compile-time dependency with a distribution dependency. You don't have a circular dependency at compile time (it wouldn't compile if you did). What you're referring to is the fact the plugins are part of the same distribution as the core of Cooja, but you want them to be distributed together (but hopefully, the core Cooja does not need plugins to compile!).

You can achieve the same easily by keeping the jars the standard way (in a repo, Maven or not). You can even call the lib folder your repo (see Gradle's repositories guide) so you can keep it in the same location you have them today. The version for the plugins may be something just nominal, but you should definitely consider versioning them for real.

I am not sure what you're referring to with "all checkouts will share a single jar in the maven-repository". That's the opposite of what I am trying to say. You should have ONE jar for each Gradle module. These jars should be in a repository (even if it's just a local directory - the difference being just how you define that location, which is a big one as it allows you to later change the repository without touching any particular dependency). Distribution is something different altogether... You can have all your jars copied into  the distribution folder later (without explicitly referring to the location of those jars - if they are in a repo, they are just normal artifacts with a coordinate you can refer to). You can even use a Ant task from Gradle as Gradle has a Groovy's AntBuilder available as the variable `ant`.

Hope it makes sense. I'm claiming this model does fit the Cooja development-model, just like this model fits any other build I can imagine.